schemas/gateway-schema.json

```diff
@@ -0,0 +1,216 @@
+                            "description": "Allow CORS on gateway.",
+                            "default": true
+                        }
+                    }
+                }
+            }
+        },
+        "server": {
+            "type": "object",
+            "description": "Configure gateway server properties.",
+            "additionalProperties": false,
+            "properties": {
+                "internal": {
+                    "type": "object",
+                    "description": "It is possible to enable another gateway port for internal communication.",
+                    "additionalProperties": false,
+                    "properties": {
+                        "enabled": {
+                            "type": "boolean",
+                            "description": "Should internal port be enabled?",
+                            "default": false
+                        },
+                        "port": {
+                            "$ref": "#/$defs/port",
+                            "default": 10017
+                        },
+                        "ssl": {
+                            "type": "object",
+                            "description": "Network encryption options for internal port. If internal is enabled, it will use Zowe SSL configuration as default.",
+                            "additionalProperties": false,
+                            "properties": {
+                                "enabled": {
+                                    "type": "boolean",
+                                    "description": "Should internal port use encryption?",
+                                    "default": false
+                                }
+                            }
+                        }
+                    }
+                },
+                "maxConnectionsPerRoute": {
+                    "type": "integer",
+                    "description": "How many connection should exists for single HTTP client?",
+                    "default": 100
+                },
+                "maxTotalConnections": {
+                    "type": "integer",
+                    "description": "How many connection should exists in total?",
+                    "default": 1000
+                },
+                "ssl": {
+                    "type": "object",
+                    "description": "Network encryption for gateway service connections.",
+                    "additionalProperties": false,
+                    "properties": {
+                        "enabled": {
+                            "type": "boolean",
+                            "description": "Should gateway use encrypted communication?",
+                            "default": true
+                        }
+                    }
+                }
+
+            }
+        },
+        "spring": {
+            "type": "object",
+            "description": "Application framework parameters",
+            "additionalProperties": false,
```

Comment:
Should we allow for other spring properties?

Reply:
removed